herd/AArch64Sem.ml

```diff
@@ -3363,7 +3363,7 @@ module Make
               fun v -> write_reg_neon_rep (neon_sz r1) r1 v ii)
         | I_FMOV_TG(_,r1,_,r2) ->
             check_neon inst;
-            !(read_reg_neon false r2 ii >>=
+            !(read_reg_neon false r2 ii >>= demote >>=
               fun v -> write_reg r1 v ii)
```

Comment:
Hi @murzin. We may seize the opportunity to perform the store to register now, resulting in one equation less:
```suggestion
            read_reg_neon false r2 ii >>= demote
             >>= fun v -> write_reg_dest r1 v ii >>= nextSet r1
```
Notice that other Neon instruction could undergo a similar "optimisation". This in particular applies to other kinds of move instructions. Nothing urgent, though: just seizing the opportunity...